src/python/pants/backend/project_info/tasks/dependencies.py

```diff
@@ -45,23 +49,43 @@ def __init__(self, *args, **kwargs):
       raise TaskError('At most one of --internal-only or --external-only can be selected.')
 
   def console_output(self, unused_method_argument):
-    ordered_closure = OrderedSet()
-    for target in self.context.target_roots:
-      if self._transitive:
-        target.walk(ordered_closure.add)
-      else:
-        ordered_closure.update(target.dependencies)
+    if self.get_options().output_format == 'json':
```

Comment:
There is a lot of duplicate work between the if/else blocks. I am a bit suprised to see different output formats require different logic when building the dep trees.

Can you break out the logic from the output format?

Reply:
I looked at it again and the issue is that the standard format output aggregates all of the dependencies into a single set without distinction of what target they belong to (when passing multiple target params). This is the cause of the logic change, and the other obvious structuring I played with ultimately ends up consolidating the iteration logic/for loops but inserts an output check in more places (whenever deciding whether you can just freely walk and aggregate multiple graph nodes or if you need to keep track/distinguish root).